### PR TITLE
Update database config

### PR DESCRIPTION
**Pull request by RandolphJiffy**
I am consolidating the database configurations spread across database.yml and .env into a single .env configuration file for easier management and development convenience in the project.

### CODE REVIEW

**Comment by RogerPodacter** [comment]
I love it!

What about `DATABASE_USERNAME` instead of `ETHS_INDEXER_DATABASE_USERNAME`?

Like it's already scoped by the app. Thoughts?

**Comment by RandolphJiffy** [comment]
Your suggestion is excellent, and I have already simplified the configuration notation.😊

File: config/database.yml [hunk 1]
```diff
@@ -19,27 +19,44 @@ default: &default
...
+  #   production:
+  #     url: <%= ENV["DATABASE_URL"] %>
+  #
+  # Read https://guides.rubyonrails.org/configuring.html#configuring-a-database
+  # for a full overview on how database connection configuration can be specified.
+  url: <%= ENV["DATABASE_URL"] %>
+
+development:
+  <<: *default
+  database: eths_indexer_development
```
Comment (RogerPodacter) [inline config/database.yml:59]: This should be pulled from ENV, no? Also maybe we call it "DATABASE_BASENAME" or something so it makes sense to append `_development` etc. Thoughts?

Comment (RandolphJiffy) [reply config/database.yml]: Your suggestion is great! 😃
I've added the database name configuration to the ENV, so everyone can name the database according to their needs.

**Comment by RandolphJiffy** [comment]
Your suggestion is great! 😃
I've added the database name configuration to the ENV, so everyone can name the database according to their needs.

File: config/database.yml [hunk 2]
```diff
@@ -19,28 +19,44 @@ default: &default
...
+  # The TCP port the server listens on. Defaults to 5432.
+  # If your server runs on a different port number, change accordingly.
+  port: <%= ENV["DATABASE_PORT"] %>
 
   # The specified database role being used to connect to PostgreSQL.
   # To create additional roles in PostgreSQL see `$ createuser --help`.
   # When left blank, PostgreSQL will use the default role. This is
   # the same name as the operating system user running Rails.
-  #username: eths_indexer
+  username: <%= ENV["DATABASE_USERNAME"] %>
```
Comment (RogerPodacter) [inline config/database.yml:35]: Will this work in the case the username and password are blank? I think this is the default setup for a local postgres. Likewise I don't think the user should have to deal with a port in most cases. Thinking about this more I'm not sure there is a one-size-fits all simplification for this config file. Maybe we should just add database.yml to gitignore and let people edit it directly.

Thoughts?

Comment (RandolphJiffy) [reply config/database.yml:35]: I've also had some new thoughts. The configuration file should cater to the needs of two types of users:

1.Novices or non-technical individuals who don't need to do much configuration to run the indexer directly.
2.Professional technical personnel who can easily fill in the configurations they need.

Additionally, there's a new consideration:
We can make the ethscriptions-indexer as a product similar to an Ethereum client, where users don't need to worry about internal implementation details. They only need to focus on the RPC interface documentation, and all data requests are made through the RPC interface.

Providing a comprehensive RPC interface and documentation seems like a good choice. This way, users don't need to learn Ruby and PostgreSQL; they can use ethscriptions-indexer within their existing technical framework.

For now, I'll temporarily close this pull request and submit it again when we have a better solution.😃😃

Comment (RogerPodacter) [reply config/database.yml:35]: What about this? https://github.com/0xFacet/ethscriptions-indexer/pull/46/files

Comment (RandolphJiffy) [reply config/database.yml:35]: > What about this? https://github.com/0xFacet/ethscriptions-indexer/pull/46/files

This is a good idea

**Pull request closed without merging**